torchdata/datapipes/iter/load/fsspec.py

```diff
@@ -133,6 +133,10 @@ def __len__(self) -> int:
         return len(self.source_datapipe)
 
 
+# Register for functional API for backward compatibility
+IterDataPipe.register_datapipe_as_function("open_file_by_fsspec", FSSpecFileOpenerIterDataPipe)
```

Comment:
As a follow-up task, add deprecation warning for the functional API.

Reply:
It seems like we will need to change the registration function and `IterDataPipe` class in PyTorch core to do that.  One possibility is [described here](https://github.com/pytorch/data/issues/480#issuecomment-1142500688).

Do you think we should land this PR as it is or make the changes to core first?

Reply:
Let's land this PR first and change the core afterwards.

Reply:
And, could you please update the deprecation table as well?